packages/application/src/connectionlost.ts

```diff
@@ -22,6 +25,45 @@ export const ConnectionLost: IConnectionLost = async function (
       'JupyterLab will continue trying to reconnect.\n' +
       'Check your network connection or Jupyter server configuration.\n'
   );
+  const buttons = [Dialog.okButton({ label: trans.__('Dismiss') })];
+
+  if (!displayConnectionLost) {
+    return;
+  }
+
+  const key = title + '----' + networkMsg;
+  const promise = connectionLostCachePromise[key];
+
+  if (promise) {
+    return promise;
+  } else {
+    try {
+      const errorDialogPromise: any = showDialog({
+        title: title,
+        body: networkMsg,
+        checkbox: {
+          label: trans.__('Do not show this message again.'),
+          caption: trans.__(
+            'If checked, you will not see a dialog informing you about an issue with server connection in the future.'
+          )
+        },
+        buttons: buttons
+      }).then(result => {
+        if (result.button.accept && !result.isChecked) {
+          delete connectionLostCachePromise[key];
+        }
+
+        if (result.isChecked) {
+          displayConnectionLost = false;
+        }
+        return;
+      });
 
-  return showErrorMessage(title, { message: networkMsg });
+      connectionLostCachePromise[key] = errorDialogPromise;
+      await errorDialogPromise;
+      return;
+    } catch (error) {
+      console.error('An error occurred while showing the dialog:', error);
```

Comment:
```suggestion
      console.error('An error occurred while showing the dialog: ', error);
```

Add space before error